activity_hub.php

```diff
@@ -496,7 +496,13 @@ function activity_descriptions()
 
     foreach ($Stage_for_id_ as $stage) {
         echo "<dt><b>$stage->id</b>: <a href='$code_url/{$stage->relative_url}'>{$stage->name}</a></dt>";
-        echo "<dd>{$stage->description}</dd>\n";
+        echo "<dd>{$stage->description}";
+        if (($stage->id == "PP") || ($stage->id == "SR")) {
+            echo _(" The counts in the Snapshot table, above, for projects available
```

Comment:
The leading space shouldn't be in the `_()` as it runs the risk of a translator removing it and it running together with the prior sentence. You probably want `echo " " . _("...");`

Reply:
I also can't remember where we are with multi-line translated strings. IIRC most of them are all in one line to avoid this?

Reply:
The problem with trying to put this part of the string in the same line (or paragraph) is that the base descriptions for all of these is in `stages.inc`, and it's not really appropriate (at least for PP) to add that string to the description. Conceivably, it might be added to the SR blurb, but I doubt the SRers care, and it would just add bloat.

We could also just ignore the issue altogether and take the additional text out. If nobody has noticed in the 16 years since the SR pool first saw light of day that projects "in SR" are also "in PP", does it really matter? Anyone who questions it, we can describe how things work.

I have no problem with taking it back out again -- I was probably over-thinking the issue to begin with :grin: -- but if we leave it in, it should stay a separate string from the rest of the blurb (and yes, if the decision is to leave it in, I'll move the leading space out of the translated string -- thanks!).

We might also consider, when we get to the point of dealing with the beginner help and the snapshot legend, whether we even need the activity descriptions on the AH, and if so, whether they should be unique instead of drawing from `stages.inc`.

Reply:
I think you misunderstood my comment. I wasn't suggesting combining it into the line above, but putting all of the translated string on a single line to be translated:
```php
echo " " . _("The counts in the Snapshot table, above, for projects available for SR are also included in the numbers of projects in PP, and the PP total.");
```

I forget out xgettext and our translation tooling handles multi-line strings.

Reply:
Ah, okay. Yes, according to the [String localization Best practices](https://www.pgdp.net/wiki/DP_Code_Best_Practices#String_localization), it should handle multi-line strings:

* You may break the string over multiple lines, so as to avoid too long lines. But if you do so, keep it a single string literal (otherwise xgettext will not pick up the correct string).

Mind you, I haven't recently experimentally verified that, but I can easily do so.

Reply:
I've now verified that it seems to work fine. It doesn't look pretty (especially in the `.po` file as viewed in a text editor) even in poedit:

<img width="503" alt="image" src="https://user-images.githubusercontent.com/36800239/132597745-a4d89657-6fe0-4d20-99c1-834dbcc76a5e.png">

But renders fine on the AH:
<img width="510" alt="image" src="https://user-images.githubusercontent.com/36800239/132597866-371224cb-841b-4baf-9c30-df70de9e487f.png">

\<cough\> ... now I should probably go back and generate a clean POT, huh?

Reply:
As long as translators know they don't need to bother with newlines or whitespaces, that's fine.